flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferReaderWriterUtil.java

```diff
@@ -187,6 +205,17 @@ static ByteBuffer allocatedHeaderBuffer() {
 		return new ByteBuffer[] { allocatedHeaderBuffer(), null };
 	}
 
+	public static ByteBuffer tryReadByteBuffer(FileChannel channel, int size) {
+		ByteBuffer bb = ByteBuffer.allocate(size);
+		try {
+			tryReadByteBuffer(channel, bb);
+			bb.flip();
+		} catch (IOException ex) {
+			ExceptionUtils.rethrow(ex);
```

Comment:
Agree, I really had the same concern before. Actually my previous way was throwing explicit `IOException` here. But I found it also need to adjust the interface method `Buffer#getNioBufferReadable` for throwing `IOException`, even it would further bring up many serious of changes for exiting usages.

It was also caused by [comment](https://github.com/apache/flink/pull/13523#discussion_r516235008). Then let's confirm them together.